src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionXpathSingleFilter.java

```diff
@@ -0,0 +1,124 @@
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.filters;
+
+import java.util.regex.Pattern;
+
+import com.puppycrawl.tools.checkstyle.TreeWalkerAuditEvent;
+import com.puppycrawl.tools.checkstyle.TreeWalkerFilter;
+import com.puppycrawl.tools.checkstyle.api.AutomaticBean;
+
+/**
+ * This filter processes {@link TreeWalkerAuditEvent}
+ * objects based on the criteria of files, checks, message, module id, xpathQuery.
+ */
+public class SuppressionXpathSingleFilter extends AutomaticBean implements
+        TreeWalkerFilter {
+    /**
+     * XpathFilter instance.
+     */
+    private XpathFilter xpathFilter;
+    /**
+     * The pattern for file names.
+     */
+    private Pattern files;
+    /**
+     * The pattern for check class names.
+     */
+    private Pattern checks;
+    /**
+     * The pattern for message names.
+     */
+    private Pattern message;
+    /**
+     * Module id of filter.
+     */
+    private String id;
+    /**
+     * Xpath query.
+     */
+    private String query;
+
+    /**
+     * Set the regular expression to specify names of files to suppress.
+     * @param files the name of the file
+     */
+    public void setFiles(String files) {
+        if (files == null) {
+            this.files = null;
+        }
+        else {
+            this.files = Pattern.compile(files);
+        }
+    }
+
+    /**
+     * Set the regular expression to specify the name of the check to suppress.
+     * @param checks the name of the check
+     */
+    public void setChecks(String checks) {
+        if (checks == null) {
+            this.checks = null;
+        }
+        else {
+            this.checks = Pattern.compile(checks);
+        }
+    }
+
+    /**
+     * Set the regular expression to specify the message of the check to suppress.
+     * @param message the message of the check
+     */
+    public void setMessage(String message) {
+        if (message == null) {
+            this.message = null;
+        }
+        else {
+            this.message = Pattern.compile(message);
+        }
+    }
+
+    /**
+     * Set the ID of the check to suppress.
+     * @param id the ID of the check
+     */
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    /**
+     * Set the xpath query.
+     * @param query the xpath query
+     */
+    public void setQuery(String query) {
+        this.query = query;
+    }
+
+    @Override
+    public boolean accept(TreeWalkerAuditEvent treeWalkerAuditEvent) {
+        return xpathFilter.accept(treeWalkerAuditEvent);
```

Comment:
@jack870131 I am working on new pitests mutations.
Please confirm atleast 1 test fails when this line is swapped with either `return true;` or `return false`.
I am not asking this code to be changed in the PR, just that we have a test that truely shows it is well tested.
If no tests fail, then you need to add atleast 1 more so it doesn't pass with this mutation.

Reply:
First new mutator will be added with https://github.com/checkstyle/checkstyle/pull/6274 .

Reply:
So should I add the codes in the testing file to show the test fail? Could you please explain more specifically? I cannot quite understand what is your meaning with adding mutator. What influence would it be with my codes or tests?

Reply:
You might have noticed some CIs named pitest. It is how we do our mutation testing. http://pitest.org/ 
It basically mutates the code in bad ways behind the scenes and expects a test to fail. It is proof that you have the needed tests for each line of code.

You don't need to add any extra documentation for this. I am just for asking for extra tests if there aren't enough already.
In this scenario, the mutator will change the code to `return true;` or `return false;` and expect a test to fail. This will show that you have a test that says this return value must be true/false in this situation and this test proves it.
If you still don't understand, I can look at it when I get time later.

Reply:
I am not sure is my test enough for this. Maybe you should check it, please. Thanks.